testing/utils/fill_cache/CMakeLists.txt

```diff
@@ -31,7 +31,7 @@ include(${rapids-cmake-dir}/cpm/spdlog.cmake)
 include(${rapids-cmake-dir}/cpm/fmt.cmake)
 include(${rapids-cmake-dir}/cpm/thrust.cmake)
 
-rapids_cpm_init()
+rapids_cpm_init(GENERATE_PINNED_VERSIONS)
```

Comment:
Will choosing to always generate the pin file affect any other tests? Also why do we need this when we should be setting this on a per-test basis for tests that call `rapids_cpm_init`

Reply:
> Will choosing to always generate the pin file affect any other tests?

No,  this only generates a pin file when we check out the local source cache of the dependencies. No effect on the tests, was done to have as much coverage on the pinning logic as possible. Not required for anything, so can remove if we choose.